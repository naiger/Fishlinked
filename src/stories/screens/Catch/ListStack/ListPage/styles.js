import { Dimensions, StyleSheet } from "react-native";

const deviceHeight = Dimensions.get("window").height;
const deviceWidth = Dimensions.get("window").width;

const styles: any = StyleSheet.create({
	container: {
		backgroundColor: "#FBFAFA",
	},
	imageContainer:{
		marginTop: -10,
		marginBottom: 10,
		width: deviceWidth * 0.9,
		height: deviceWidth * 0.7,
		alignItems: 'center',
		justifyContent: 'center',
	},
	image: {
		borderWidth: 1,
		borderColor: "#ddd",
		width: deviceWidth * 0.7,
		height: deviceWidth * 0.7,
		alignItems: 'center',
		justifyContent: 'center',
	},
	searchContainer:{
		alignItems: 'center',
	},
	seatchbar:{
		marginTop: 5,
		height: deviceHeight / 20,
		width: deviceWidth * 0.9,
		backgroundColor: "#ddd",
	},
});
export default styles;
