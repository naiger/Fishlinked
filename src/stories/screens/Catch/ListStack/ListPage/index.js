// @flow
import styles from "./styles";
import { Container, Header, Left, Button, Icon, Body, Title, Right, Content, Item, Input, Spinner, ListItem, CardItem, Card, Thumbnail, List, Text, Segment} from "native-base";
import * as React from "react";
import {Image,View, Alert, RefreshControl} from "react-native";

export interface Props {
  navigation: any;
  //State
  isLoading: bool,
  list: Array<Object>,
  error: Object,
  searchResult:Array<Object>,
  searchCatchKeyword:string,
  catchListIsSearching:boolean,
  userIsValid: boolean,
  myCatch: boolean,
  //Functions
  searchCatches: Function,
  isMyCatch: Function,
}

export interface State {}
class ListPage extends React.Component<Props, State> {
  render() {
    let list = [];
		this.props.searchCatchKeyword === ""
		? list = this.props.list
		: list= this.props.searchResult
    return (
      <Container style={styles.container}>
        <Header hasSegment>
          <Left>
          {
            this.props.myCatch
            ? (<Button transparent onPress={() => {
                this.props.isMyCatch(false);
                this.props.navigation.navigate("MePage");
              }}>
                <Icon active name="arrow-back"/>
              </Button>)
            : null
          }
          </Left>
          <Body><Title>{this.props.myCatch ? "My Catch" : "Catch"}</Title></Body>
          <Right>
            <Button transparent onPress={() => {
              this.props.userIsValid
              ? this.props.navigation.navigate("PhotoBrowser")
              : Alert.alert(
								 "Alert",
								 "You must login first to access this page.",
								 [
									 {text: "Login", onPress: () => this.props.navigation.navigate("LoginPage",{from: "ListPage", to:"PhotoBrowser"})},
									 {text: "Sign up", onPress: () => this.props.navigation.navigate("RegisterPage")},
									 {text: "Cancel", style:"cancel"},
								 ]
               )
            }}>
              <Icon active name="camera"/>
            </Button>
          </Right>
        </Header>
        <View style={styles.searchContainer}>
          <Item inlineLable rounded style={styles.seatchbar}>
            <Icon name="ios-search" />
            <Input
              placeholder="Search"
              onChangeText={text => this.props.searchCatches(text)}/>
           </Item>
        </View>
        <Content
          refreshControl={
            <RefreshControl
              refreshing={this.props.isLoading || this.props.catchListIsSearching}
              onRefresh={()=>this.props.searchCatches("")}
              enabled={true}
              titile="Loading..."/>
        }>
          <List>
          {
            list && list.length > 0
            ? list.map((item, i) => (
              <ListItem
                key={i}
                onPress={() =>{this.props.navigation.navigate("CatchPage", {catch: item});}}
              >
              <Card>
                <CardItem>
                  <Left>
                    <Thumbnail source={{uri:'Image URL'}} />
                    <Body>
                      <Text>{item.title.rendered}</Text>
                      <Text>{item.date.slice(0,10)}</Text>
                    </Body>
                  </Left>
                </CardItem>
                <CardItem cardBody>
                  <View style={styles.imageContainer}>
                  {
                    item.better_featured_image != null
                    ? <Image style={styles.image} source={{uri: item.better_featured_image.source_url}}/>
                    : <View style={styles.image}><Text>No Picture</Text></View>
                  }
                  </View>
                </CardItem>
              </Card>
              </ListItem>
            ))
            : this.props.isLoading
              ? null
              : <Text>No Result</Text>
          }
          </List>
        </Content>
      </Container>
    );
  }
}

export default ListPage;
