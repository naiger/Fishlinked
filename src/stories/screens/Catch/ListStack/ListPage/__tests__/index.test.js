import React from "react";
import ListPage from "../index";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

const navigation = { navigate: jest.fn() };
const list = { map: jest.fn() };

it("renders correctly", () => {
	const tree = renderer.create(<ListPage navigation={navigation} list={list} />).toJSON();
	expect(tree).toMatchSnapshot();
});
