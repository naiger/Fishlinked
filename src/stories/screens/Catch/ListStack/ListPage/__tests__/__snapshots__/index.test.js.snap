// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FBFAFA",
      },
    ]
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutoAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    onScroll={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={0}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    virtual={undefined}
  >
    <View>
      <View
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
        virtual={undefined}
      />
    </View>
  </RCTScrollView>
</View>
`;
